client/src/modules/list/components/Items/model/reducer.js

```diff
@@ -75,18 +75,18 @@ const items = (state = {}, action) => {
       const {
         payload: { description, isOrdered, name, editedBy, _id: itemId }
       } = action;
-      const previousItem = state[itemId];
-      let previousDescription;
 
-      if (previousItem) {
-        previousDescription = previousItem.description;
-      }
+      const previousItem = state[itemId];
+      const previousDescription = previousItem.description;
+      const newDescription = isDefined(description)
+        ? description
+        : previousDescription;
 
       return {
         ...state,
         [itemId]: {
           ...(previousItem || action.payload),
-          description: description || previousDescription,
+          description: newDescription,
```

Comment:
hmm, I think about something like this: 

this are our function in helpers:
```
const defined = x => x !== undefined;

const filter = f => object =>
  fromEntries(
    Object.entries(object).filter(([key, value]) => f(value, key, object))
  );

```
I think about those additional helpers:
```
const updated = x => definded(x) && x !== '';
export const filterUpdated = filter(updated);


```
and in reducer:

```
const { _id: itemId, ...rest } = action.payload;
const dataToUpdate = filterUpdated(rest);

return {...state, [itemId]: { ...state[itemId, ...dataToUpdate }
```
If we'll add another prop to Item and this prop will be editable, nothing will change and we won't have to check another prop manually... (I hope so)

Reply:
Yes, it looks nice but did you tested this with a single value coming?
`{name: 'test'} `??

Because I've been trying to use filterUndefined() helper function, but values returned from that funcitons were weird like` {1: {n}, 2:{a}, 3:{m}, 4: {e}}`

Reply:
it works in cohort case:
```
const cohorts = (state = {}, action) => {
  switch (action.type) {
    case CohortActionTypes.CREATE_SUCCESS:
      return { ...state, [action.payload._id]: { ...action.payload } };
    case CohortActionTypes.UPDATE_SUCCESS: {
      const { cohortId, ...data } = action.payload;
      const previousCohort = state[cohortId];
      const dataToUpdate = filterDefined(data);

      const updatedCohort = {
        ...previousCohort,
        ...dataToUpdate
      };

      return { ...state, [cohortId]: updatedCohort };
    }
    case CohortActionTypes.ARCHIVE_SUCCESS:
```

Reply:
yes it works because there is always an object with data comming witn name and description.
so it filters object perfectly. But if the string is passed it doesnt work as expected. 

If you want me to use it, test it

Reply:
and in locks case: 
```
case CohortHeaderStatusTypes.UNLOCK: {
      const {
        payload: { cohortId, locks }
      } = action;
      const { locks: prevLocks } = state[cohortId];
      const updatedLocks = filterDefined(locks);

      return {
        ...state,
        [cohortId]: {
          ...state[cohortId],
          locks: { ...prevLocks, ...updatedLocks }
        }
      };
    }
```

Reply:
Yes, you're right, it's alway an object. You merged many methods to UpdateItem and after this merge reducer became too much expanded (especially in 404. branch) I would prefer always send an object and have simple and clean reducer but decide and resolve comments